DXRando/DeusEx/Classes/DXRCigaretteMachine.uc

```diff
@@ -0,0 +1,18 @@
+class DXRCigaretteMachine injects CigaretteMachine;
+
+function Frob(Actor frobber, Inventory frobWith)
+{
+    local int usesBefore;
+    local DXRando dxr;
+
+    usesBefore = numUses;
+
+    Super.Frob(frobber, frobWith);
+
+    if (numUses == 0 && usesBefore != 0) {
```

Comment:
Might as well add the other vending machine events too, VendingMachineDispense_Cigs and VendingMachineDispense, for consistency